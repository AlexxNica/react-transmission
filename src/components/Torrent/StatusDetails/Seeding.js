import React from 'react';
import { FormattedMessage } from 'react-intl';

import { formatUL } from 'util/formatters';

export default ({ torrent: { peersGettingFromUs, peersConnected, rateUpload } }) => (
  <FormattedMessage
    id='torrent.full.seeding'
    defaultMessage={`Seeding to {peersGettingFromUs, number} of {peersConnected, number} connected {peersConnected, plural, one{peer} other{peers}} - {formattedRateUpload}`}
    values={{
      peersGettingFromUs,
      peersConnected,
      formattedRateUpload: formatUL(rateUpload),
    }}
  />
);
