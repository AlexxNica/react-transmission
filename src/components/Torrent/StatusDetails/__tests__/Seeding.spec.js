import React from 'react';
import createComponentWithIntl from 'test/util/createComponentWithIntl';

import Seeding from '../Seeding';

test('Seeding none', () => {
  const component = createComponentWithIntl(
    <Seeding torrent={{peersGettingFromUs: 0, peersConnected: 0, rateUpload: 0}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Seeding single peers', () => {
  const component = createComponentWithIntl(
    <Seeding torrent={{peersGettingFromUs: 1, peersConnected: 1, rateUpload: 1E5}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('Seeding multiple peers', () => {
  const component = createComponentWithIntl(
    <Seeding torrent={{peersGettingFromUs: 5, peersConnected: 5, rateUpload: 1E5}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});
