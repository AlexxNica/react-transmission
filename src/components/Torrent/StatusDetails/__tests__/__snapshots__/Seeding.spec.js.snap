exports[`test Seeding multiple peers 1`] = `
<span>
  Seeding to 5 of 5 connected peers - ↑ 100 kB/s
</span>
`;

exports[`test Seeding none 1`] = `
<span>
  Seeding to 0 of 0 connected peers - ↑ 0 kB/s
</span>
`;

exports[`test Seeding single peers 1`] = `
<span>
  Seeding to 1 of 1 connected peer - ↑ 100 kB/s
</span>
`;
